packages/salesforcedx-vscode-apex/src/index.ts

```diff
@@ -315,6 +312,24 @@ async function createLanguageClient(extensionContext: vscode.ExtensionContext) {
 
     languageClientUtils.setClientInstance(languageClient);
     await languageClient!.start();
+    // Client is running
+
+    // The listener should be set after languageClient is ready
+    // Language client will get notified once async init jobs are done
+    languageClient.onNotification('indexer/done', async () => {
+      await getTestOutlineProvider().refresh();
+      languageServerReady();
+    });
+
+    // client will check status with server after initialization
+    const indexerStatus = await languageClient.sendRequest('indexer/status');
+    if (indexerStatus === ServiceState.RUNNING) { // 2 is RUNNING state
+      await getTestOutlineProvider().refresh();
+      languageServerReady();
+    } else if ([ServiceState.STOPPING, ServiceState.FAILED, ServiceState.TERMINATED].includes(indexerStatus as number)) {
```

Comment:
@mingxuanzhangsfdx failures in language server startup or shutting down are handled in [apexErrorhandler.ts](https://github.com/forcedotcom/salesforcedx-vscode/blob/develop/packages/salesforcedx-vscode-apex/src/apexErrorHandler.ts). Can you tell me why you felt it necessary to cover these two states here as well?

Reply:
My guts are that there is no callback from indexer service in sync mode, so language server has no idea if indexer service is healthy after start it async'ly. The situation that language server is ready but indexer failed is not what users want.